apps/ui/components/reservation-unit/QuickReservation.tsx

```diff
@@ -168,552 +136,175 @@ const SlotButton = styled.button`
-    return null;
-  }
-
-  const startDay = dayMax([new Date(interval.startDatetime), minDay]) ?? minDay;
-  const daysToGenerate = reservationsMaxDaysBefore ?? 180;
-  const days = Array.from({ length: daysToGenerate }, (_, i) =>
-    addDays(startDay, i)
-  );
-
-  // Find the first possible time for that day, continue for each day until we find one
-  for (const day of days) {
-    const availableTimesForDay = getAvailableTimesForDay({
-      ...props,
-      day,
-      fromStartOfDay: true,
-      reservationUnit,
-    });
-    if (availableTimesForDay.length > 0) {
-      const [hours, minutes] = availableTimesForDay[0].split(":").map(Number);
-      day.setHours(hours, minutes, 0, 0);
-      return day;
-    }
-  }
-
-  return null;
-};
-
 const QuickReservation = ({
-  isSlotReservable,
-  isReservationUnitReservable,
-  createReservation,
   reservationUnit,
-  setErrorMsg,
   subventionSuffix,
-  shouldUnselect,
-  setInitialReservation,
-  quickReservationSlot,
-  calendarRef,
-  setQuickReservationSlot,
-  apiBaseUrl,
+  reservationForm,
+  focusSlot,
+  durationOptions,
+  startingTimeOptions,
+  nextAvailableTime,
+  submitReservation,
+  LoginAndSubmit,
 }: Props): JSX.Element | null => {
-  const { t, i18n } = useTranslation();
-
-  const {
-    minReservationDuration,
-    maxReservationDuration,
-    reservationStartInterval,
-  } = reservationUnit || {};
-
-  const durationOptions = useMemo(() => {
-    if (
-      minReservationDuration == null ||
-      maxReservationDuration == null ||
-      reservationStartInterval == null
-    ) {
-      return [];
-    }
-    return getDurationOptions(
-      minReservationDuration ?? undefined,
-      maxReservationDuration ?? undefined,
-      reservationStartInterval,
-      t
-    );
-  }, [
-    minReservationDuration,
-    maxReservationDuration,
-    reservationStartInterval,
-    t,
-  ]);
-
-  // TODO this should be on a timer
-  const nextHour: Date = useMemo(() => {
-    const now = new Date();
-    now.setMinutes(0);
-    now.setSeconds(0);
-    now.setMilliseconds(0);
-    now.setHours(now.getHours() + 1);
-    return now;
-  }, []);
-
-  const [localReservation, setLocalReservation] =
-    useState<ReservationProps | null>(null);
-  const [date, setDate] = useState(() => {
-    const result = new Date();
-    result.setHours(0, 0, 0, 0);
-    return result;
-  });
-  const [time, setTime] = useState<string>(
-    formatDate(nextHour.toISOString(), "HH:mm")
-  );
-  const [duration, setDuration] = useState<OptionType | undefined>(
-    durationOptions.find((n) => n.value === "1:00") ?? durationOptions[0]
-  );
-  const [slot, setSlot] = useState<string | null>(null);
-  const [isReserving, setIsReserving] = useState(false);
+  const { t } = useTranslation();
+  const { setValue, watch, handleSubmit } = reservationForm;
+  const formDate = watch("date");
+  const dateValue = useMemo(() => new Date(formDate ?? ""), [formDate]);
```

Comment:
This seems off? is what's the Date format?

Reply:
Yup, same thing as in ReservationCalendarControls